src/main/java/in/projecteka/datanotificationsubscription/SecurityConfiguration.java

```diff
@@ -0,0 +1,222 @@
+        httpSecurity
+                .authorizeExchange()
+                .pathMatchers(GATEWAY_APIS).hasAnyRole(GATEWAY.name())
+                .pathMatchers("/**")
+                .authenticated();
+        return httpSecurity
+                .authenticationManager(authenticationManager)
+                .securityContextRepository(securityContextRepository)
+                .build();
+    }
+
+    @Bean
+    public ReactiveAuthenticationManager authenticationManager() {
+        return new AuthenticationManager();
+    }
+
+    @Bean({"jwtProcessor"})
+    public ConfigurableJWTProcessor<com.nimbusds.jose.proc.SecurityContext> getJWTProcessor() {
+        return new DefaultJWTProcessor<>();
+    }
+
+    @Bean("internalServiceAuthenticator")
+    public CMTokenAuthenticator internalServiceAuthenticator(
+            @Qualifier("identityServiceJWKSet") JWKSet jwkSet,
+            CacheAdapter<String, String> blockListedTokens,
+            ConfigurableJWTProcessor<com.nimbusds.jose.proc.SecurityContext> jwtProcessor) {
+        return new CMTokenAuthenticator(jwkSet, blockListedTokens, jwtProcessor);
+    }
+
+    @Bean
+    public SecurityContextRepository contextRepository(
+            GatewayTokenVerifier gatewayTokenVerifier,
+            @Value("${subscriptionmanager.authorization.header}") String authorizationHeader,
+            @Qualifier("internalServiceAuthenticator") CMTokenAuthenticator cmTokenAuthenticator) {
+        return new SecurityContextRepository(
+                gatewayTokenVerifier,
+                authorizationHeader,
+                cmTokenAuthenticator);
+    }
+
+    @AllArgsConstructor
+    private static class SecurityContextRepository implements ServerSecurityContextRepository {
+        private final GatewayTokenVerifier gatewayTokenVerifier;
+        private final String authorizationHeader;
+        private final CMTokenAuthenticator cmTokenAuthenticator;
+
+        @Override
+        public Mono<Void> save(ServerWebExchange exchange, org.springframework.security.core.context.SecurityContext context) {
+            throw new UnsupportedOperationException("No need right now!");
+        }
+
+        @Override
+        public Mono<org.springframework.security.core.context.SecurityContext> load(ServerWebExchange exchange) {
+            var requestPath = exchange.getRequest().getPath().toString();
+            var requestMethod = exchange.getRequest().getMethod();
+
+            if (isAllowedList(requestPath)) {
+                return empty();
+            }
+
+            if (isGatewayAuthenticationOnly(requestPath, requestMethod)) {
+                return checkGateway(exchange.getRequest().getHeaders().getFirst(AUTHORIZATION))
+                        .switchIfEmpty(error(unAuthorized()));
+            }
+
+            var token = exchange.getRequest().getHeaders().getFirst(authorizationHeader);
+            token = addBearerIfNotPresent(token);
```

Comment:
We should have `isEmpty` check before adding `Bearer`? Otherwise it will never be empty.